### PR TITLE
fix: incoming call keeps showing up (AR-2934)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2934" title="AR-2934" target="_blank">BugAR-2934</a>  Call keeps on coming back after answering on desktop client
  </td></table>
  <br />
 

observe incomingCallUseCase instead of periodic check

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

A follow up PR of wireapp/kalium#1327  to prevent the incoming screen from showing.

### Issues

Extra work done to get incoming calls, which could lead to performance issue

### Solutions

Simplicity it by observing the incomingCalls flow instead of creating a new flow every time and invoke the usecase

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Check you are able to receive calls

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by vitorhugods — APPROVED** [review]
👨‍🍳👌

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 1]
```diff
@@ -141,36 +140,13 @@ class WireNotificationManager @Inject constructor(
         // TODO: for now GetIncomingCallsUseCase() doesn't return valid data on the first try.
         //      so it's possible to have scenario, when FCM comes informing us that there is a Call,
         //      but we don't get it from the first GetIncomingCallsUseCase() call.
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt]: If this is no longer needed, we can remove it.

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1325/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1325/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5533028)(5533028)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv0.0.1(5533028)(5533028)-staging-compat.apk)

**Pull request merged**